packages/synapse-react-client/src/components/GoalsV2/GoalsV2.test.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react'
+import { render, screen, waitFor } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import GoalsV2 from './GoalsV2'
+import {
+  SynapseContextProvider,
+  SynapseContextType,
+} from '../../utils/context/SynapseContext'
+
+const mockSynapseContext: Partial<SynapseContextType> = {
+  isInExperimentalMode: false,
+  utcTime: false,
+  withErrorBoundary: false,
+  downloadCartPageUrl: '/DownloadCart',
+  appId: undefined,
+}
+
+const queryClient = new QueryClient()
+
+const server = setupServer(
+  rest.post(
```

Comment:
Your handlers look good, but we have a utility `generateAsyncJobHandlers` that can abstract away setting up the `async/start` and `async/get` handlers that might reduce the amount of code here. Optionally consider using that here